service/frontend/workflowHandler.go

```diff
@@ -1860,6 +1867,10 @@ func (wh *WorkflowHandler) GetWorkflowExecutionHistory(
 		getRequest.MaximumPageSize = common.Int32Ptr(common.GetHistoryMaxPageSize)
 	}
 
+	if wh.historyArchived(ctx, getRequest, domainID) {
```

Comment:
This results into extra call to history service for every GetWorkflowExecutionHistory invocation.  It would be good to consolidate this behind under single GetMutableState call.

Reply:
Did not address will explain offline why.